### PR TITLE
Add support for Decimal Constant Vector

### PR DESCRIPTION
**Pull request by majetideepak**

### CODE REVIEW

File: velox/vector/tests/VectorTest.cpp [hunk 1]
```diff
@@ -1275,9 +1269,18 @@ TEST_F(VectorTest, compareNan) {
 class VectorCreateConstantTest : public VectorTest {
  public:
   template <TypeKind KIND>
-  void testPrimitiveConstant(typename TypeTraits<KIND>::NativeType val) {
+  void testPrimitiveConstant(
+      typename TypeTraits<KIND>::NativeType val,
+      const TypePtr& type = {}) {
```
Comment (mbasmanova) [inline velox/vector/tests/VectorTest.cpp:1274]: perhaps, change the default to `Type::create<KIND>()` and simplify expectedStr logic below:

```
auto expectedStr = fmt::format(
        "[CONSTANT {}: {} value, {} size]",
        type->toString(),
        baseVector->toString(0),
        size_);
```

**Review by mbasmanova — APPROVED** [review]
@majetideepak Looks good % small comment.

**Comment by majetideepak** [comment]
@mbasmanova can we merge this? Thanks.

**Comment by majetideepak** [comment]
@pedroerp, @mbasmanova any update on this PR? Thanks.

**Comment by mbasmanova** [comment]
I had to rebase the diff and the flaky internal test failed again. I'm re-running it now and will try to land again.

**Pull request closed without merging**